python/cudf/cudf/core/column/column.py

```diff
@@ -1684,32 +1691,45 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
                     dtype = pd.api.types.pandas_dtype(dtype)
                     if is_categorical_dtype(dtype):
                         raise TypeError
-                    else:
-                        np_type = np.dtype(dtype).type
-                        if np_type == np.bool_:
-                            pa_type = pa.bool_()
-                        else:
-                            pa_type = np_to_pa_dtype(np.dtype(dtype))
+
+                pa_data = pa.array(
+                    arbitrary,
+                    type=dtype.pa_type if dtype is not None else None,
+                    from_pandas=True if nan_as_null is None else nan_as_null,
+                )
+                # todo: fix this ???? ????????
+                as_column_dtype = (
+                    cudf.dtype(pa_data.type)
+                    if not isinstance(
+                        pa_data.type, (pa.lib.DictionaryType, pa.lib.ListType)
+                    )
+                    else None
+                )
                 data = as_column(
-                    pa.array(
-                        arbitrary,
-                        type=pa_type,
-                        from_pandas=True
-                        if nan_as_null is None
-                        else nan_as_null,
-                    ),
-                    dtype=dtype,
-                    nan_as_null=nan_as_null,
+                    pa_data, dtype=as_column_dtype, nan_as_null=nan_as_null
                 )
+
             except (pa.ArrowInvalid, pa.ArrowTypeError, TypeError):
                 if is_categorical_dtype(dtype):
-                    sr = pd.Series(arbitrary, dtype="category")
-                    data = as_column(sr, nan_as_null=nan_as_null, dtype=dtype)
-                elif np_type == np.str_:
+                    if (
+                        isinstance(dtype, pd.CategoricalDtype)
+                        or dtype is "category" # noqa: F632
```

Comment:
It's because it wants us to use `== /  !=` for string comparisons. But the way we have our `__eq__` set up in `CategoricalDtype`, `==` works with just the word 'category' and we actually don't want it to here.